packages/api/src/services/context/context-service-registry.ts

```diff
@@ -0,0 +1,84 @@
+import type { ContextService } from './context.service';
+import {Service} from '../../providers/service/service';
+import {ValueChangeCallback} from '../../models/context/value-change-callback';
+import {BeforeChangeValidationPromise} from '../../models/context/before-change-validation-promise';
+
+/**
+ * Describes a global subscription to value changes in any registered ContextService.
+ * This is used to apply the same callback logic to all context-managed properties across services.
+ */
+type GlobalSubscription = {
+  callback: ValueChangeCallback<unknown>;
+  beforeChangeValidationPromise?: BeforeChangeValidationPromise<unknown>;
+  afterChangeCallback?: ValueChangeCallback<unknown>;
+};
+
+/**
+ * ContextServiceRegistry is responsible for tracking all registered ContextService instances in the application.
+ * It supports global subscription to context changes and notifies newly registered services of existing global listeners.
```

Comment:
I don't quite understand this doc. Tracking services? For what? And what is a global subscription?

Reply:
fixed